docs/docs/shards/types.md

```diff
@@ -0,0 +1,642 @@
+
+For example, in [`(Math.Mean)`](https://docs.fragcolor.xyz/shards/Math/Mean/) the value for `:Kind` parameter needs to be of type **Enum**.
+
+```clojure linenums="1"
+(Math.Mean
+  :Kind [(Enum)]
+)
+```
+
+`(Math.Mean)` computes the mean of a sequence of floating-point numbers. But there are three kinds of means - Arithmetic mean, Geometric mean, and Harmonic mean.
+
+So the parameter `:Kind` is defined as an enum variable with these three fixed states :
+` :Kind = {Arithmtic mean, Geometric mean, Harmonic mean}`
+
+And hence `:Kind` expects a value that matches one of its possible states. In other words the value you pass in for `:Kind` needs to be an enumerated data type.
+
+In simple terms it just means that you pass in one of the allowed named constant values. Anything else will fail validation.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Arithmetic)
+    (Log)   ;; AM => 6
+
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Geometric)
+    (Log)   ;; GM => 4.47214
+
+    [2.0 10.0]
+    (Math.Mean :Kind Mean.Harmonic)
+    (Log)   ;; HM => 3.33333
+    
+    [2.0 10.0]
+    (Math.Mean 
+    ;; AM is default, anything else will throw error
+        ;; :Kind 123
+        )   
+    (Log)   ;; AM => 6
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 15:35:00.868] [T-15316] [logging.cpp::55] [mywire] 6
+    [info] [2022-07-22 15:35:00.881] [T-15316] [logging.cpp::55] [mywire] 4.47214
+    [info] [2022-07-22 15:35:00.882] [T-15316] [logging.cpp::55] [mywire] 3.33333
+    [info] [2022-07-22 15:35:00.883] [T-15316] [logging.cpp::55] [mywire] 6
+    ```
+
+## Float
+
+Type **Float** defines a 64-bit signed floating point number.
+
+!!! note
+    Has keyword [`float`](https://docs.fragcolor.xyz/functions/values/#float) and alias `Float`.
+
+Floating point means it has the capability to store a decimal point and hence supports decimal numbers.
+
+64 bits of memory allows this data type to support a very large range of positive and negative decimal numbers (16 significant decimal digits and an exponent range of −383 to +384).
+
+A **Float** value looks like this: `(float 2.53)`.
+It may also be represented without the keyword `float`, with just the floating-point value: `2.53`.
+
+[`(Math.Add)`](https://docs.fragcolor.xyz/shards/Math/Add/) is an example of a shard that uses this data type for its input, output, and `:Operand` parameter.
+
+=== "Code"
+
+    ```clojure linenums="1"
+    (float 2.4) (Math.Add (float 1.43))
+    (Log)   ;; float output => 3.83
+
+    2.4 (Math.Add 1.43)
+    (Log)   ;; float output => 3.83
+    ```
+
+=== "Output"
+
+    ```
+    [info] [2022-07-22 22:06:32.856] [T-20204] [logging.cpp::55] [mywire] 3.83
+    [info] [2022-07-22 22:06:32.873] [T-20204] [logging.cpp::55] [mywire] 3.83
+    ```
+
+## Float2
+
+Type **Float2** defines a vector of two [**Float**](#float) type numbers.
```

Comment:
Should we call them *tuple* instead of *vector*?

cc @sinkingsugar @guusw

Reply:
It's actually a vector. Also vector is a concept much easier to understand.